bin/imsim.py

```diff
@@ -98,24 +100,24 @@ def main():
                               m5=[defaults.m5(bandpass)],
                               seeing=[seeing])
 
-    # Trim the input catalog to a single chip.
-    raICRS = phoSimObjectList['ra'].values
-    decICRS = phoSimObjectList['dec'].values
-    phoSimObjectList['chipName'] = chipNameFromRaDec(raICRS, decICRS,
-                                                     camera=camera,
-                                                     obs_metadata=obs,
-                                                     epoch=2000.0)
     # Now further sub-divide the source dataframe into stars and galaxies.
     if arguments.sensor is not None:
+        # Trim the input catalog to a single chip.
+        raICRS = phoSimObjectList['ra'].values
+        decICRS = phoSimObjectList['dec'].values
+        phoSimObjectList['chipName'] = chipNameFromRaDec(raICRS, decICRS,
+                                                         camera=camera,
+                                                         obs_metadata=obs,
+                                                         epoch=2000.0)
         starDataBase = \
-            phoSimObjectList.query("galSimType == 'pointSource' and magNorm < 50 and chipName=='%s'" % arguments.sensor)
+            phoSimObjectList.query("galSimType=='pointSource' and magNorm<50 and chipName=='%s'" % arguments.sensor)
```

Comment:
We aren't doing infinite precision math. If we pick a magnitude value that's large enough, it will underflow to zero eventually.   That point probably is closer to 1000 than it is to 50, logarithmically speaking, but since our zero points seem to be around ~30-33, I figured 50 would be fine.